python_modules/dagster/dagster/_core/storage/event_log/in_memory.py

```diff
@@ -43,17 +41,27 @@ def _create_connection(self):
         SqlEventLogStorageMetadata.create_all(conn)
         alembic_config = get_alembic_config(__file__, "sqlite/alembic/alembic.ini")
         stamp_alembic_rev(alembic_config, conn)
-        return engine, conn
+
+        self._engine = engine
+        self._conn = conn
+        self.reindex_events()
+        self.reindex_assets()
 
     @contextmanager
     def run_connection(self, run_id=None):
+        if not self._conn:
+            self._create_connection()
         yield self._conn
 
     @contextmanager
     def index_connection(self):
+        if not self._conn:
+            self._create_connection()
         yield self._conn
```

Comment:
maybe we should just hold on to the engine and do context managed `connects` per call and not hold on to the connection

Reply:
I think we need to hold the connection, since the db is only consistent within a connection, which is normally fine for an in-memory use case.